source/slang/slang-check-conversion.cpp

```diff
@@ -201,11 +201,137 @@ namespace Slang
         return baseStructDeclRef;
     }
 
+    bool SemanticsVisitor::_hasExplicitConstructor(StructDecl* structDecl)
+    {
+        bool result = false;
+        for (auto ctor : getMembersOfType<ConstructorDecl>(getASTBuilder(), structDecl, MemberFilterStyle::All))
+        {
+            // constructor that is not synthesized must be user defined.
+            if (ctor.getDecl()->findModifier<SynthesizedModifier>() == nullptr)
+            {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    // TODO: We might need to find a good way to get the synthesized constructor instead of traversing all of constructors.
```

Comment:
You can have a `SLANG_UNREFLECTED` on the structdecl to cache the result of this function.